### ISSUE TITLE
Infinite wait when using Chromium installed via Snap

### ISSUE THREAD

**Issue by kristianLundkvist** [HollowMan6/mdbook-pdf#12]
Current behavior: Ubuntu have decided to install Chromium using Snap and this causes the PDF generation to quietly fail. Unless you pass arguments to get a more detailed log it will simply display `Generating PDF, please be patient...` until the process is manually killed. Using debian packages resolves the issue, [this method to be precise](https://askubuntu.com/questions/1204571/how-to-install-chromium-without-snap).

Expected behavior: Since the issue isn't really caused by mdbook-pdf, a note in the README about it and how to manually resolve it would suffice. A bonus would be some way of detecting that chrome failed to start properly, displaying this to the user with a link to the issue and the solution but this might not be feasible.

Logs:

Generated using `RUST_BACKTRACE=full RUST_LOG=trace mdbook build`, shortened to the relevant part.

```log
Generating PDF, please be patient...
[2023-02-15T10:24:20Z TRACE headless_chrome::browser::process] Chrome will have profile: --user-data-dir=/tmp/rust-headless-chrome-profilexjCsX9
[2023-02-15T10:24:20Z INFO  headless_chrome::browser::process] Launching Chrome binary at "/snap/bin/chromium"
[2023-02-15T10:24:20Z INFO  headless_chrome::browser::process] Started Chrome. PID: 40118
[2023-02-15T10:24:20Z TRACE headless_chrome::browser::process] Chrome output: ERROR: ld.so: object '/usr/local/lib/AppProtection/libAppProtection.so' from /etc/ld.so.preload cannot be preloaded (failed to map segment from shared object): ignored.
[2023-02-15T10:24:20Z TRACE headless_chrome::browser::process] Chrome output: ERROR: ld.so: object '/usr/local/lib/AppProtection/libAppProtection.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
```

**Comment by HollowMan6** [comment]
Hi! Thanks for reporting this. Unfortunately this looks like an issue with mdbook-pdf's dependency [rust-headless-chrome](https://github.com/atroche/rust-headless-chrome), as I can't see any issue related to snap at their side and actually chromium has already been released as a snap package for a long time, I would assume that this is related to this issue as well: https://github.com/atroche/rust-headless-chrome/issues/367 I'll try to bump the version of our [rust-headless-chrome](https://github.com/HollowMan6/mdbook-pdf-headless_chrome) to see if it can get fixed.

By the way, I would like to mention that infinite wait won't happen, as I just extend the wait time to 10 min, so that it would also work in the CI for generating some very large PDF books. It would eventually fail in 10 min.

**Comment by HollowMan6** [comment]
I can reproduce your issue with current version of mdbook-pdf under chrome 110, even it's not installed by snap. Will work on this now.

**Comment by HollowMan6** [comment]
That's interesting. Technically I would say that it would be something related to [rust-headless-chrome](https://github.com/atroche/rust-headless-chrome), and I'm afraid I can't help you more. I have already updated rust-headless-chrome to the latest version, and I can't find an issue related to snap chromium at their side. I'll try to reproduce this and meanwhile you can [open an issue there](https://github.com/atroche/rust-headless-chrome/issues) to see if there's some kind of solution.

**Comment by HollowMan6** [comment]
Hi! I don't think there's any issue using the snap chromium with mdbook-pdf. I have tested the snap chromium to generate pdf file using the [vagrant file](https://github.com/HollowMan6/mdbook-pdf/blob/main/Vagrantfile) I just committed to the master branch and everything works well. I would assume that it's because something related to your snapd is broken, and I will close this issue since I can't reproduce this.
